service/history/ndc/workflow_state_replicator.go

```diff
@@ -678,6 +678,8 @@ func (r *WorkflowStateReplicatorImpl) getNewRunMutableState(
 		return nil, err
 	}
 
+	newRunMutableState.InitTransitionHistory()
```

Comment:
`getNewRunMutableState` is also used by event based replication when replicating closed workflows. I think a more appropriate place to add this is within `getNewRunWorkflow` function which is only used by state based code path.

Reply:
updated using isStateBased flag.